pontoon/base/tests/managers/test_entity.py

```diff
@@ -928,3 +928,67 @@ def test_mgr_entity_reset_active_translations(resource_a, locale_a):
     active = entities[4].translation_set.filter(active=True)
     assert active[0].string == entities[4].string + " translation1"
     assert active[1].string == entities[4].string_plural + " translation1plural"
+
+
+@pytest.mark.django_db
+def test_get_word_count_exclude_non_words(resource_a, locale_a):
+    """ How many words in a string
+    """
+    testEntitiesQuerySet = Entity.for_project_locale(resource_a.project, locale_a)
+    count = testEntitiesQuerySet._get_word_count(
+        "String 123 =+& string hh-gg object.string"
+    )
+    assert count == 5
+
+
+@pytest.mark.django_db
+def test_get_word_count_simple(resource_a, locale_a):
+    """ How many words in a string
+    """
+    testEntitiesQuerySet = Entity.for_project_locale(resource_a.project, locale_a)
+    count = testEntitiesQuerySet._get_word_count("There are 7 words in this string")
+    assert count == 7
+
+
+@pytest.mark.django_db
+def test_mgr_get_or_create(resource_a, locale_a):
+    """
+    Get or create entities method works and counts words.
+    """
+    testEntitiesQuerySet = Entity.for_project_locale(resource_a.project, locale_a)
+    arguments = {
+        "resource": resource_a,
+        "string": "simple string",
+    }
+    obj, created = testEntitiesQuerySet.get_or_create(**arguments)
+
+    assert created
+    assert obj.word_count == 2
+
+
+@pytest.mark.django_db
+def test_mgr_bulk_update(resource_a, locale_a):
+    """
+    Update entities method.
+    """
+    objs = [
+        EntityFactory.create(resource=resource_a, string="testentity %s" % i,)
```

Comment:
@mathjazz I have a question here. In case I'm using EntityFactory to create a list of entities for bulk_update, I receive an error: 

> AttributeError: 'list' object has no attribute 'pk'

How I can fix this? May I just assign to pk attribute numbers '1,2,3 etc' during EntityFactory.create() ?

Reply:
When the tests fails, you should see values of the relevant variables, which show that the `update_fields = None` and that the value of `meta` is what should be in `update_fields`:
```
objs = [<Entity: testentity 0>, <Entity: testentity 1>], meta = ['resource', 'string', 'string_plural', 'key', 'comment', 'group_comment', ...], update_fields = None, exclude_fields = None
using = 'default', batch_size = 2, pk_field = 'pk'
```

So we have a bug in https://github.com/mozilla/pontoon/pull/1495/files#diff-c18ce63ddb8aca155c0553a0c1b174e3R2315 - you should return `bulk_update` with keyword arguments, not positional:
```python
return bulk_update(objs, update_fields=update_fields, batch_size=batch_size)
```

That's for proving how useful the unit tests are! :-)

Reply:
> you should return bulk_update with keyword arguments, not positional:

Oh, I see now.